pkg/bootstrap/platform/aws.go

```diff
@@ -105,18 +125,32 @@ func getAWSInfo(path string, ipv6 bool) (string, error) {
 }
 
 // getRegion returns the Region that the instance is running in.
-func getRegion(ipv6 bool) string {
-	region, _ := getAWSInfo("placement/region", ipv6)
+func getRegion(ipv6 bool, headers map[string]string) string {
+	region, _ := getAWSInfo("placement/region", ipv6, headers)
 	return region
 }
 
 // getAvailabilityZone returns the AvailabilityZone that the instance is running in.
-func getAvailabilityZone(ipv6 bool) string {
-	az, _ := getAWSInfo("placement/availability-zone", ipv6)
+func getAvailabilityZone(ipv6 bool, headers map[string]string) string {
+	az, _ := getAWSInfo("placement/availability-zone", ipv6, headers)
 	return az
 }
 
-func getInstanceID(ipv6 bool) string {
-	instance, _ := getAWSInfo("instance-id", ipv6)
+func getInstanceID(ipv6 bool, headers map[string]string) string {
+	instance, _ := getAWSInfo("instance-id", ipv6, headers)
 	return instance
 }
+
+func getToken(ipv6 bool) string {
+	url := awsMetadataTokenIPv4URL
+	if ipv6 {
+		url = awsMetadataTokenIPv6URL
+	}
+
+	resp, err := http.DoHTTPGetWithTimeout(url, time.Millisecond*100)
```

Comment:
This should be an HTTP PUT, not an HTTP GET. It is also missing a required header:
> The PUT request must include a header that specifies the time to live (TTL) for the token, in seconds, up to a maximum of six hours (21,600 seconds). The token represents a logical session. The TTL specifies the length of time that the token is valid and, therefore, the duration of the session.

General question: why implement all of this logic instead of using the AWS SDK?

Reply:
Full sample request:
```
TOKEN=`curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600"` \
&& curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/
```

Reply:
yes, I misread the code, wait for a dev env to test this code.

> General question: why implement all of this logic instead of using the AWS SDK?

will increase the size of binary too much for simple usecase.

Reply:
@adam-bloom can you help to verify this? I need more time setup my env on AWS.

Reply:
@zirain i haven’t contributed to istio development before - can you point me in the right direction to validate this? Is there a way to exercise this code path via `go run` or something simple, or do I need to build a sidecar image and use that in my k8s cluster somewhere? I have access to a development AWS environment and a k8s/EKS cluster with istio if we need it - happy to put that to use. Just need a little guidance to make it approachable.

Reply:
@adam-bloom please add following configuration into you demo app, and check the logs in container istio-proxy.

```
annotations:
  "sidecar.istio.io/agentLogLevel": "default:debug"
  "sidecar.istio.io/proxyImage": "ghcr.io/zirain/proxyv2:zirain-dev"
```

Reply:
Ah, looking closer, the first one is the google cloud detection. I see a different user agent now. If that fails, we should still be ok. I can try enforcing IMDS v2 to extra confirm if you’d like, but I think this change is good.

Reply:
@adam-bloom, thanks for your help.

It will be wonderful if you can enforce IMDSv2 and verify if everything is OK.
My account is still on the way, will be ok tomorrow.